service/routers/users.js

```diff
@@ -26,20 +27,40 @@ const getUsers = async (req, res) => {
 }
 
 const getUsersById = async (req, res) => {
-  const user = await User.findById(req.params.id)
-  res.send(user)
+  const requestor = await User.findById(userId)
+  const { userId } = req.user
+  if (requestor.role === 'admin' || requestor.role === 'superuser') {
+    const user = await User.findById(req.params.id)
+  } else if(requestor.role === 'user' && requestor.id === user.id){
+    res.send(sanitizerUsers(users))
```

Comment:
What's the purpose behind this function? The logic behind the two code paths doesn't make sense